sql: panic in SPLIT AT

Found with RSG:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: ALTER TABLE COVERING SPLIT AT ( PLACEHOLDER ): runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: ALTER TABLE COVERING SPLIT AT ( PLACEHOLDER ): runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x6e82b5]

goroutine 56 [running]:
panic(0x15a07c0, 0xc4208c7330)
        /mnt/kena/go1.7rc5/src/runtime/panic.go:500 +0x1a1
github.com/cockroachdb/cockroach/util/stop.(*Stopper).Recover(0xc4201b6000)
        /mnt/kena/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:179 +0x6e
panic(0x15a07c0, 0xc4208c7330)
        /mnt/kena/go1.7rc5/src/runtime/panic.go:458 +0x243
github.com/cockroachdb/cockroach/sql.(*Executor).ExecuteStatements.func1(0xc420c6c69c, 0x2d)
        /mnt/kena/go/src/github.com/cockroachdb/cockroach/sql/executor.go:426 +0x149
panic(0x15bf7c0, 0xc420012160)
        /mnt/kena/go1.7rc5/src/runtime/panic.go:458 +0x243
github.com/cockroachdb/cockroach/sql.(*planner).Split(0xc4208b4890, 0xc4209903c0, 0xc42011ab40, 0x63, 0x63, 0xc42011ab80)
        /mnt/kena/go/src/github.com/cockroachdb/cockroach/sql/split.go:41 +0xb35
github.com/cockroachdb/cockroach/sql.(*planner).newPlan(0xc4208b4890, 0x7fccd4078c28, 0xc4209903c0, 0x0, 0x0, 0x0, 0x1bb25201, 0x244c900, 0x162c7, 0x0, ...)
        /mnt/kena/go/src/github.com/cockroachdb/cockroach/sql/plan.go:311 +0x148d
github.com/cockroachdb/cockroach/sql.(*planner).makePlan(0xc4208b4890, 0x7fccd4078c28, 0xc4209903c0, 0x1, 0xc4208d5150, 0x1, 0x1, 0xc42116c4e8)
        /mnt/kena/go/src/github.com/cockroachdb/cockroach/sql/plan.go:217 +0x70
github.com/cockroachdb/cockroach/sql.(*Executor).execStmt(0xc4202680f0, 0x7fccd4078c28, 0xc4209903c0, 0xc4208b4890, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /mnt/kena/go/src/github.com/cockroachdb/cockroach/sql/executor.go:1146 +0xc2
github.com/cockroachdb/cockroach/sql.(*Executor).execStmtInOpenTxn(0xc4202680f0, 0x7fccd4078c28, 0xc4209903c0, 0xc4208b4890, 0xa60101, 0xc4208b4828, 0x0, 0x0, 0x0, 0x0, ...)
        /mnt/kena/go/src/github.com/cockroachdb/cockroach/sql/executor.go:1035 +0x26c
github.com/cockroachdb/cockroach/sql.(*Executor).execStmtsInCurrentTxn(0xc4202680f0, 0xc4208d50a0, 0x1, 0x1, 0xc4208b4890, 0xc4208b4828, 0xc421160101, 0xc42116d150, 0x0, 0x0, ...)
        /mnt/kena/go/src/github.com/cockroachdb/cockroach/sql/executor.go:778 +0x246
github.com/cockroachdb/cockroach/sql.runTxnAttempt(0xc4202680f0, 0xc4208b4890, 0x1, 0xc4208b4828, 0xc4208d50f0, 0xc4208d50a0, 0x1, 0x1, 0x57cea149, 0x1bb23385, ...)
        /mnt/kena/go/src/github.com/cockroachdb/cockroach/sql/executor.go:699 +0xeb
github.com/cockroachdb/cockroach/sql.(*Executor).execRequest.func2(0xc4210b89a0, 0xc4208d50f0, 0x4000000000000000, 0x0)
        /mnt/kena/go/src/github.com/cockroachdb/cockroach/sql/executor.go:583 +0x20a
github.com/cockroachdb/cockroach/internal/client.(*Txn).Exec(0xc4210b89a0, 0xc420730101, 0xc420156e40, 0xc420733680, 0x0, 0x0)
        /mnt/kena/go/src/github.com/cockroachdb/cockroach/internal/client/txn.go:559 +0x20a
```

--- a/sql/split.go
+++ b/sql/split.go
@@ -27,27 +27,34 @@ import (
 // Split executes a KV split.
 // Privileges: INSERT on table.
 func (p *planner) Split(n *parser.Split) (planNode, error) {
-	var index sqlbase.IndexDescriptor
-	var tableDesc *sqlbase.TableDescriptor
+	var tableName parser.NormalizableTableName
 	if n.Index == nil {
-		tn, err := n.Table.NormalizeWithDatabaseName(p.session.Database)
-		if err != nil {
-			return nil, err
-		}
-		tableDesc, err = p.getTableDesc(tn)
-		if err != nil {
-			return nil, err
-		}
+		tableName = n.Table
+	} else {
+		tableName = n.Index.Table
+	}
+
+	// Check that the table exists and that the user has permission.
+	tn, err := tableName.NormalizeWithDatabaseName(p.session.Database)
+	if err != nil {
+		return nil, err
+	}
+	tableDesc, err := p.getTableDesc(tn)
+	if err != nil {
+		return nil, err
+	}
+	if tableDesc == nil {
+		return nil, sqlbase.NewUndefinedTableError(tn.String())
+	}
+	if err := p.checkPrivilege(tableDesc, privilege.INSERT); err != nil {
+		return nil, err
+	}
+
+	// Determine which index to use.
+	var index sqlbase.IndexDescriptor
+	if n.Index == nil {
 		index = tableDesc.PrimaryIndex
 	} else {
-		tn, err := n.Index.Table.NormalizeWithDatabaseName(p.session.Database)
-		if err != nil {
-			return nil, err
-		}
-		tableDesc, err = p.getTableDesc(tn)
-		if err != nil {
-			return nil, err
-		}
 		normIdxName := sqlbase.NormalizeName(n.Index.Index)
 		status, i, err := tableDesc.FindIndexByNormalizedName(normIdxName)
 		if err != nil {
@@ -58,9 +65,8 @@ func (p *planner) Split(n *parser.Split) (planNode, error) {
 		}
 		index = tableDesc.Indexes[i]
 	}
-	if err := p.checkPrivilege(tableDesc, privilege.INSERT); err != nil {
-		return nil, err
-	}
+
+	// Determine how to use the remaining argument expressions.
 	if len(index.ColumnIDs) != len(n.Exprs) {
 		return nil, errors.Errorf("expected %d expressions, got %d", len(index.ColumnIDs), len(n.Exprs))
 	}
@@ -77,6 +83,7 @@ func (p *planner) Split(n *parser.Split) (planNode, error) {
 		}
 		typedExprs[i] = typedExpr
 	}
+
 	return &splitNode{
 		p:         p,
 		tableDesc: tableDesc,

--- a/sql/testdata/alter_table
+++ b/sql/testdata/alter_table
@@ -447,3 +447,9 @@ Field Type Null  Default
 a     INT  false NULL
 b     INT  true  NULL
 c     INT  true  NULL
+
+statement error table "nonexistent" does not exist
+ALTER TABLE nonexistent SPLIT AT (42)
+
+statement error table "nonexistent" does not exist
+ALTER INDEX nonexistent@noindex SPLIT AT (42)

